test_stopwatch.py

```diff
@@ -70,6 +73,34 @@ def test_stopwatch_cancel_context_manager(self):
         assert len(agg_values) == 2
         assert all([span in agg_values for span in ('root', 'root#grand')])
 
+        # Ensure that we are not leaking cancelled span data
+        assert not sw._cancelled_spans
+
+    def test_stopwatch_full_cancel(self):
+        """Test that an entire span - from root to children, can be cancelled."""
```

Comment:
Were you looking to add a test here. It appears there is an empty test. Otherwise the rest looks good.